### PR TITLE
Implement nn_estimator evaluating method in Orca Bigdl estimator

### PR DESCRIPTION
**Pull request by leonardozcm**
Related issue: #3507
When call `estimator.evaluate` to deal with a dataFrame, it will call scala method `internalEval` to process data.

Usage:
```python
 result = est.evaluate(data=validationDF, batch_size=16, feature_cols="image")
```
As a result it returns a Jlist of EvaluatedResult, which will be converted to a python dict like:
```
2021-03-16 00:04:27 INFO  DistriOptimizer$:1759 - Top1Accuracy is Accuracy(correct: 8, count: 26, accuracy: 0.3076923076923077)
{'Top1Accuracy': 0.3076923191547394}
```
Working Progress:
- [x] Scala internalEval method
- [x] Python nn_estimator eval method
- [x] UT

And if there is any spark official evaluating trait or base class you know, pls leave your comments, thanks!

### CODE REVIEW

File: zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala [hunk 1]
```diff
@@ -477,6 +489,30 @@ class NNEstimator[T: ClassTag] private[zoo] (
     wrapBigDLModel(model)
   }
 
+  def internalEval(dataFrame: DataFrame): JList[EvaluatedResult] = {
+    val validationFeatureset = getDataSet(dataFrame, validationBatchSize)
```
Comment (yangw1234) [inline zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala:493]: I think you can directly call `model.evaluate`.

Comment (leonardozcm) [reply zoo/src/main/scala/com/intel/analytics/zoo/pipeline/nnframes/NNEstimator.scala:493]: synced offline

**Comment by leonardozcm** [comment]
http://10.239.47.210:18888/job/ZOO-PR-Validation/5200/

File: pyzoo/zoo/pipeline/nnframes/nn_classifier.py [hunk 2]
```diff
@@ -473,6 +490,12 @@ def setLabelCol(self, value):
         """
         return self._set(labelCol=value)
 
+    def eval(self, val_data):
```
Comment (yangw1234) [inline pyzoo/zoo/pipeline/nnframes/nn_classifier.py:493]: `NNEstimator` is public API, I think we can fisrt move this method (and the other three set methods) to somewhere else and not expose them to the user of `NNEstimator`.

Comment (leonardozcm) [reply pyzoo/zoo/pipeline/nnframes/nn_classifier.py]: have change to `_eval`

**Comment by leonardozcm** [comment]
http://10.239.47.210:18888/job/ZOO-PR-Python-ExampleTests/2434/

File: pyzoo/zoo/pipeline/nnframes/nn_classifier.py [hunk 3]
```diff
@@ -384,6 +383,24 @@ def getValidation(self):
         """
         return self.validation_config
 
+    def setNNBatchSize(self, batch_size):
```
Comment (yangw1234) [inline pyzoo/zoo/pipeline/nnframes/nn_classifier.py:386]: also make these methods private

Comment (yangw1234) [reply pyzoo/zoo/pipeline/nnframes/nn_classifier.py:386]: and add some comments documenting why you need these methods

Comment (leonardozcm) [reply pyzoo/zoo/pipeline/nnframes/nn_classifier.py:386]: Sure

**Comment by yangw1234** [comment]
let's first merge it after jenkins passed. We can look into whether we can use pyspark.ml.evaluator later.

**Comment by leonardozcm** [comment]
http://10.239.47.210:18888/job/ZOO-PR-Validation/5225/

**Pull request merged**